bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/TypeUtils.java

```diff
@@ -159,4 +160,10 @@ public static Type getReferredType(Type type) {
         return constraint;
     }
 
+    public static Type getEffectiveType(Type type) {
```

Comment:
Does this have to be added to the public API?

Reply:
We can move this to internal package since we do not have requirement to expose.